frontend/src/components/pages/CreateReview/BookModal.tsx

```diff
@@ -0,0 +1,275 @@
+import { CheckIcon, SmallCloseIcon } from "@chakra-ui/icons";
+import {
+  Button,
+  FormControl,
+  FormLabel,
+  Grid,
+  GridItem,
+  Modal,
+  ModalBody,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Stack,
+} from "@chakra-ui/react";
+import React, { useState } from "react";
+
+import { BookFormats, Genre } from "../../../constants/Enums";
+import { Author, Book, BookFormat, Publisher } from "../../../types/BookTypes";
+import AddNumberInput from "./AddNumberInput";
+import AddSelect from "./AddSelect";
+import AddStringInput from "./AddStringInput";
+import AddStringInputList from "./AddStringInputList";
+
+interface BookModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+  booksAdded: Book[];
+  handleBooksAdded: (b: Book[]) => void;
+}
+
+const BookModal = (props: BookModalProps): React.ReactElement => {
```

Comment:
Right now, when I add the book and the modal closes (or when I click cancel to close the modal), the next time I open the modal, my previous field entries are still populated. Can we make it so that I get a blank modal? Once you have added a book, it doesn't make much sense to have the old data pre-populate if I try to add a new book etc right?

Reply:
Thanks for pointing this out, the behaviour has been adjusted so that data is wiped whenever the modal is closed, whether it's through saving the modal, or by clicking outside the modal